src/validation.cpp

```diff
@@ -1595,47 +1636,121 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
         }
     }
 
-    auto multi_submission_result = quit_early || linearized_package.FilteredTxns().empty() ?
-        PackageMempoolAcceptResult(package_state_quit_early, {}) :
-        AcceptSubPackage(linearized_package.FilteredTxns(), args);
-    PackageValidationState& package_state_final = multi_submission_result.m_state;
+    // Clear temporary coins that were pulled in during PreChecks or created in PackageAddTransaction.
+    CleanupTemporaryCoins();
+
+    // Now that we have determined which transactions to skip, do full validation for each
+    // transaction with its ancestor subpackage. If this is a single transaction, call
+    // AcceptSingleTransaction() with a single transaction ATMPArgs to avoid unnecessary package
+    // restrictions like disallowment of RBF. Skip when we know the feerate will not be sufficient.
+    // If we encounter any errors at all, we quit immediately and do not validate anything else.
+    // This means we may potentially skip validation for valid transactions, but we can expect that
+    // some other peer will announce those transactions to us.
```

Comment:
Not sure if I understand what you mean fully, but the point here is to try to avoid an attack where somebody tries to block a transaction by stuffing invalid transactions into a package with it. Nodes will still be `inv`ing transactions meeting the peer's fee filter so, unless there is a CPFP or other missing inputs situation, the receiver won't need to download this transaction as a package. The model here is we assume we have at least 1 peer who is honest / isn't actively creating invalid packages.

For a LN commitment transaction + CPFP child, they'll announce the child, you'll download it, request the ancestor package, and then validate commitment transaction + child. I don't expect that there will be any other transactions in that package. Even if there is another descendant on the second anchor output or a grandchild, your node will still be told the existence of the CPFP child, and still be told the 2-transaction ancestor package if you request it.

> blocking an in-package LN transaction propagation by submitting something conflicting to the victim's peers

This seems equivalent to a single transaction scenario. If all your peers have tx A which conflicts with tx B, then you probably won't hear about B. I don't think there's anything we can do about that here...